File: goalbot-frontend/src/types/task.ts
export interface Task {
  id: number
  userId: number
  goalId?: number
  goalTitle?: string
  title: string
  description?: string
  planDate: string
  startTime?: string
  endTime?: string
  plannedMinutes: number
  status: number
  createdAt?: string
  updatedAt?: string
}

export interface TaskPayload {
  goalId?: number
  title: string
  description?: string
  planDate: string
  startTime?: string
  endTime?: string
  plannedMinutes: number
  status: number
}

export interface TaskQuery {
  date?: string
  goalId?: number
  status?: number
}

export interface IcsImportEvent {
  uid?: string
  title: string
  description?: string
  location?: string
  planDate: string
  startTime?: string
  endTime?: string
  plannedMinutes: number
  allDay: boolean
  skipped: boolean
  skipReason?: string
}

export interface IcsImportResult {
  sourceEventCount: number
  expandedEventCount: number
  importedCount: number
  skippedCount: number
  dryRun: boolean
  events: IcsImportEvent[]
  importedTasks: Task[]
  warnings: string[]
}

export interface IcsImportOptions {
  dryRun?: boolean
  startDate?: string
  endDate?: string
  skipExisting?: boolean
}

File: goalbot-backend/src/main/java/com/example/goalbot/entity/Task.java
package com.example.goalbot.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

@Data
@TableName("task")
public class Task {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long userId;

    private Long goalId;

    private String title;

    private String description;

    private LocalDate planDate;

    private LocalTime startTime;

    private LocalTime endTime;

    private Integer plannedMinutes;

    private Integer status;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;
}
